changelog.d/sc-549.fixed

```diff
@@ -0,0 +1 @@
+Fixed bug where Semgrep would fail to generate unreachable findings for a particular dependency if a reachable finding for that dependency had already been found in the same lockfile, even if there are multiple copies of this dependency at different vulnerable versions.
```

Comment:
```suggestion
Fixed bug where Semgrep would fail to generate unreachable findings for a particular library if a reachable finding for that library had already been found in the same lockfile, even if there are multiple copies of this library at different vulnerable versions.
```
maybe I'd phrase it like this, I always thought of `dependency` as `(library, version)` and that's why I was confused; this is why I think you can have the same library as a direct dependency and as a transitive dependency at the same time